maven-plugin/src/main/java/org/openmrs/maven/plugins/Pull.java

```
@@ -0,0 +1,293 @@
+                wizard.showMessage(String.format(UPDATING_MODULE_MESSAGE, project.getArtifactId()));
+                if (pullLatestUpstream(project.getPath())) {
+                    wizard.showMessage(String.format(PULL_SUCCESS_MESSAGE, project.getArtifactId()));
+                } else {
+                    getLog().error(String.format(PULL_ERROR_MESSAGE, project.getArtifactId(), failureReason));
+                    getLog().error(UPDATE_MANUALLY_MESSAGE);
+                    pullingError = true;
+                }
+            }
+            if(pullingError){
+                throw new MojoExecutionException("Some modules could not be updated automatically, please see error messages above");
+            }
+        }else {
+            wizard.showMessage(String.format(NO_WATCHED_MODULES_MESSAGE, serverId));
+        }
+    }
+
+    /**
+     * Pull latest changes from upstream master
+     *
+     * @param path
+     * @return true if pull was successful
+     * @throws MojoExecutionException
+     */
+    private boolean pullLatestUpstream(String path) throws MojoExecutionException {
+        Repository localRepo;
+        Git git = null;
+        String previousBranch;
+        String newBranchFull;
+        String newBranch = "sdk-" + serverId;
+        try {
+            localRepo = new FileRepository(new File(path, ".git").getAbsolutePath());
+            git = new Git(localRepo);
+
+            previousBranch = localRepo.getBranch();
+
+            checkoutAndCreateNewBranch(git, newBranch);
+            newBranchFull = localRepo.getFullBranch();
+            RevCommit stash = git.stashCreate().call();
+            addRemoteTempRepo(git, path, newBranch);
+            PullResult pullResult = pullFromRemote(git, newBranch);
+            if(!pullResult.isSuccessful()){
+                rebaseAbort(git);
+                checkoutBranch(git, previousBranch);
+                failureReason = CONFLICT_MESSAGE;
```

It's better to throw a checked exception with the given message instead of having failureReason.